src/main/java/com/sap/cloudfoundry/client/facade/domain/RouteDestination.java

```diff
@@ -0,0 +1,30 @@
+package com.sap.cloudfoundry.client.facade.domain;
+
+import java.util.UUID;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.sap.cloudfoundry.client.facade.Nullable;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableRouteDestination.class)
+@JsonDeserialize(as = ImmutableRouteDestination.class)
+public abstract class RouteDestination extends CloudEntity implements Derivable<RouteDestination> {
+
+    public abstract UUID getApplicationGuid();
+
+    @Nullable
+    public abstract Integer getPort();
+
+    @Nullable
+    public abstract Integer getWeight();
+
+    public abstract String getProtocol();
```

Comment:
Shouldn't this be nullable as well? We can return null in the "raw" object when building `RouteDestination`

Reply:
_Protocol to use for this destination. Valid protocols are http1 or http2 if route protocol is http, tcp if route protocol is tcp. A null value will set it either http1 or tcp based on the route protocol; this feature is only available if you turn it on in routing release_
According to the doc, it should not return null. We use the object only for getting a route, for the creation of the destination requestedProtocol is used